### DISCUSSION TITLE
Impossible to use java serialization mechanism ?

### DISCUSSION THREAD

**Discussion by legzo** [ExpediaGroup/graphql-kotlin#1480] [Q&A]
Hi,

We would like to cache some data returned by back-end services with redis and java serialization. The fact that the classes generated from the schema do not implement `java.io.Serializable` make this impossible. Is there a way to customize the generator to add the `Serializable` implementation ?

Thanks !

**Comment by dariuszkuc** [comment]
Hello 👋 
It is not possible. Currently only `Jackson` and `kotlinx-serialization` formats are supported.

**Comment by samuelAndalon** [comment]
in Addition to what @dariuszkuc mentioned, you can also write as many indirection layers as you want to achieve the caching

```kotlin
data class User(val name: String, val lastName: String)

class UserService(
    userRedisRepository: UserRedisRepository
) {
    fun getUser(id: Int): CompletableFuture<User> {
        userRedisRepository.find(id) ?: // make your sevice call
    }
}

class SimpleQuery(
  val userService = UserService()
) : Query {
    suspend fun user(id: Int): User = userService.getUser(id).await()
}
```

**Reply by legzo** [reply]
Hi and thank you for your answer.

I'm not quite sure what you mean. In your example `User` can implement `Serializable` so that I can store it into redis and the equivalent of the `User` class, generated from the GraphQL schema would not. So I have to map the GraphQL object to my serializable class, right ?
